circular.go

```diff
@@ -193,12 +193,20 @@ func (buf *Buffer) Write(p []byte) (int, error) {
 	return n, nil
 }
 
+// capacity returns the current buffer capacity.
+func (buf *Buffer) capacity() int {
```

Comment:
```suggestion
func (buf *Buffer) cap() int {
```

I think `cap` is something you expect from Go tho :) And it's not a keyword.

Reply:
I would rather not call it `cap` so that it's not confused with `cap(data)`